build: allow bazel build ...

alexeagle opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

alexeagle commented:
Note, the reason this PR removes `firebase-tools` is:

1) firebase-tools has an optional dependency on https://www.npmjs.com/package/@google-cloud/functions-emulator
2) yarn's `--ignore-optional` doesn't work for transitive deps, so there's no way to yarn install without getting that functions-emulator package
3) functions-emulator has a transitive dep on `grpc`
4) the version of `grpc` we get has `BUILD` files and no `WORKSPACE` file so it always breaks `bazel build ...`

It could be solved by any of:
1) remove firebase-tools - this is what I did
2) fix yarn so you can omit optional deps of a transitive dep
3) make functions-emulator depend transitively on a more recent `grpc` version
4) patch `grpc` after install by doing an `rm` command in our postinstall or something

gkalpak commented:
Maybe one could use [resolutions][1] to pin `grpc` to a new version (as long as it is compatible with `functions-emulator` - or maybe we don't care :grin:)

[1]: https://yarnpkg.com/en/docs/package-json#toc-resolutions

alexeagle commented:
Thanks @gkalpak I'll try that if Igor doesn't like this approach

IgorMinar commented on package.json:
```diff
@@ -63,7 +63,6 @@
     "cors": "2.8.4",
     "domino": "1.0.29",
     "entities": "1.1.1",
-    "firebase-tools": "3.12.0",
```
this change has nothing to do with the `bazel build ...` commit. Please move to a separate commit, or remove in favor of #22186

alexeagle replied:
like I explained above, this is *required* for `bazel build ...` - it's the main thing that was in the way

alexeagle replied:
copied the explanation to the commit message

IgorMinar reviewed (approved):
the rest looks great. thanks

IgorMinar reviewed (approved):
please amend the commit message with explanation about the protobuffjs

vicb commented:
This PR caused master to go red:
- firebase is required to publish the build artifacts size ATM,
- bumping zone.js seems to be a pb (needs to be investiagted)

alexeagle committed:
build: allow bazel build //...

Also switch our CircleCI commands to just
bazel build //...
bazel test //...
as this is easier to understand.

Note, the reason this commit removes `firebase-tools` is:

1) firebase-tools has an optional dependency on
https://www.npmjs.com/package/@google-cloud/functions-emulator
2) yarn's `--ignore-optional` doesn't work for transitive deps, so
there's no way to yarn install without getting that functions-emulator
package
3) functions-emulator has a transitive dep on `grpc`
4) the version of `grpc` we get has `BUILD` files and no `WORKSPACE`
file so it always breaks `bazel build ...`

It could be solved by any of:
1) remove firebase-tools - this is what I did
2) fix yarn so you can omit optional deps of a transitive dep
3) make functions-emulator depend transitively on a more recent `grpc`
version
4) patch `grpc` after install by doing an `rm` command in our
postinstall or something

In its place we must install protobufjs. This is needed by the
ngc-wrapped test, which needs jasmine as well as bazel's worker mode
dependencies, and therefore cannot simply rely on
node_modules =
"@build_bazel_rules_typescript_tsc_wrapped_deps//:node_modules"

matsko commented:
@alexeagle please rebase onto master.

alexeagle commented:
Rebased
